utils/log_request.ts

```diff
@@ -0,0 +1,44 @@
+import { HttpContext } from '@adonisjs/core/http'
+import { Logger } from '@adonisjs/core/logger'
+import onFinised from 'on-finished'
+
+/* v8 ignore start */
+const getLogLevel = (statusCode: number) => {
+  if (statusCode < 400) return 'info'
+  if (statusCode >= 400 && statusCode < 500) return 'warn'
+  
+  return 'error'
+}
+/* v8 ignore end */
+
+const log = (
+  logger: Logger,
+  url: string,
+  method: string,
+  statusCode: number,
+  startedAt?: bigint
+) => {
+  const logLevel = getLogLevel(statusCode)
+
+  const strings = [method, statusCode, url]
+  if (startedAt) {
+    const ms = Number(process.hrtime.bigint() - startedAt) / 1e6
+    strings.push(`${ms.toFixed(1)}ms`)
+  }
+
+  const message = strings.join(' ')
+
+  /* v8 ignore next */
+  if (process.env.NODE_ENV !== 'test') logger[logLevel](message)
```

Comment:
We don't want our test logs to be full of http request logging. So we skip the print step for tests, but exercise all the the remaining logging code.